spring-boot-admin-server-ui/src/main/frontend/views/instances/details/mem-chart.vue

```diff
@@ -23,73 +23,64 @@
   />
 </template>
 
-<script>
+<script setup lang="ts">
 import moment from 'moment';
 import prettyBytes from 'pretty-bytes';
-import { useI18n } from 'vue-i18n';
+import { computed } from 'vue';
 
 import LineChart from '@/views/instances/details/LineChart';
 
-export default {
-  components: { LineChart },
-  props: {
-    data: {
-      type: Array,
-      default: () => [],
+const { data } = defineProps<{
+  data: Array<any>;
+}>();
+
+const datasets = computed(() => {
+  const hasMetaspace = Object.values(data).some((d) => d.metaspace !== null);
+
+  const _datasets = {
+    used: {
+      label: 'instances.details.memory.used',
     },
-  },
-  setup(props) {
-    const { t } = useI18n();
-    return { ...props, t };
-  },
-  data() {
-    return {
-      chart: undefined,
-      label: 'timestamp',
-      datasets: {
-        used: {
-          label: 'instances.details.memory.used',
-        },
-        metaspace: {
-          label: 'instances.details.memory.metaspace',
+    committed: {
+      label: 'instances.details.memory.committed',
+    },
+  };
+
+  if (hasMetaspace) {
+    _datasets.metaspace = {
+      label: 'instances.details.memory.metaspace',
+    };
+  }
+
+  return _datasets;
+});
+
+const config = {
+  options: {
+    plugins: {
+      tooltip: {
+        callbacks: {
+          title: (ctx) => {
+            return prettyBytes(ctx[0].parsed.y);
+          },
+          label: (ctx) => {
+            return this.t(ctx.dataset.label);
```

Comment:
where does t come from now?

Reply:
Leftover when migrating from options to composition API.